Update the new auto-complete-config behavior (#4632)
triton-inference-server/server: docs/model_configuration.md

on Triton. In some cases the required portions of the model
configuration can be generated automatically by Triton. The
required portion of the model configuration are the settings
shown in the [Minimal Model Configuration](#minimal-model-configuration).
By default, Triton will try to complete these sections. However,
by starting Triton with `--disable-auto-complete-config` option,
Triton can be configured to use the provided configuration as is.

Triton can derive all the required settings automatically for
most of the TensorRT, TensorFlow saved-model, and ONNX models.
For Python models, [`auto_complete_config`](https://github.com/triton-inference-server/python_backend/#auto_complete_config)
function can be implemented in Python backend to provide
[`max_batch_size`](#maximum-batch-size), [`input`](#inputs-and-outputs)
and [`output`](#inputs-and-outputs) properties using `set_max_batch_size`,
`add_input`, and `add_output` functions. These properties will allow Triton
to load the Python model with [Minimal Model Configuration](#minimal-model-configuration)
in absence of a configuration file. 
All other model types *must* provide a model configuration file.

> A custom backend developer can populate required settings in the

Reviewer: ```suggestion
When developing a custom backend you populate required settings in the
```

configuration and call `TRITONBACKEND_ModelSetConfig` API to
update completed configuration with Triton core. You can take a
look at [TensorFlow](https://github.com/triton-inference-server/tensorflow_backend)
and [Onnxruntime](https://github.com/triton-inference-server/onnxruntime_backend)

> backends as examples of how to acheive this. The users would still

Reviewer: ```suggestion
backends as examples of how to acheive this. You would still
```

> be required to provide a config.pbtxt with `backend` field for these

Reviewer: ```suggestion
need to provide a config.pbtxt with `backend` field for these
```

> custom backends.

Reviewer 2: " The users *must* 
provide a config.pbtxt with `backend` field for
custom backends."

> custom backends.

Reviewer: It might be worth mentioning that all the other fields except inputs, outputs, max_batch_size, and batching settings will be ignored if it is specified by the custom backend.
